### PR TITLE
Add testing of new query overload to the perf benchmark

### PR DESCRIPTION
**Pull request by aprokop**
The callbacks simply count the number of found neighbors.

The PR should be pretty straightforward. One thing that we need to decide on is whether we want to test both types of queries on the same number of points for performance testing. Testing both on everything essentially doubles testing time.

If decide to not test the same way, the answer will depend on what we think is more important.
I think my current position would be to test all numbers for pure callbacks, and not more than 3 for the old callbacks (with indices and offsets).

### CODE REVIEW

File: benchmarks/bvh_driver/bvh_driver.cpp [hunk 1]
```diff
@@ -51,9 +54,12 @@ struct Spec
     return s;
   }
 
-  std::string create_label_radius_search(std::string const &tree_name) const
+  std::string create_label_radius_search(std::string const &tree_name,
+                                         std::string const &flavor = "") const
```
Comment (masterleinad) [inline benchmarks/bvh_driver/bvh_driver.cpp:58]: I would also get rid of the default argument here.

**Review by masterleinad — COMMENTED** [review]
Apparently, this is failing our performance benchmark test: https://code.ornl.gov/ecpcitest/alexa/-/jobs/753819.
Apart from that, this looks acceptable to me.

**Comment by aprokop** [comment]
> Apparently, this is failing our performance benchmark test: [code.ornl.gov/ecpcitest/alexa/-/jobs/753819](https://code.ornl.gov/ecpcitest/alexa/-/jobs/753819).

Hmm, weird, and hard to see why.

File: benchmarks/bvh_driver/bvh_driver.cpp [hunk 2]
```diff
@@ -40,9 +40,12 @@ struct Spec
     return s;
   }
 
-  std::string create_label_knn_search(std::string const &tree_name) const
+  std::string create_label_knn_search(std::string const &tree_name,
+                                      std::string const &flavor = "") const
```
[the remark this answers is not in the document]
Comment (aprokop) [reply benchmarks/bvh_driver/bvh_driver.cpp:44]: No, there is a place that calls it with default argument.

File: benchmarks/bvh_driver/bvh_driver.cpp (hunk 1 above)
Comment (aprokop) [reply benchmarks/bvh_driver/bvh_driver.cpp:58]: No, there is a place that calls it with default argument.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Comment by aprokop** [comment]
Running benchmarks took 45 minutes on Ascent.

**Pull request merged**